frontend/integration-tests/tests/monitoring.scenario.ts

```diff
@@ -193,32 +183,197 @@ describe('Monitoring: Silences', () => {
+    expect(firstElementByTestID('pagerduty-receiver-form').isPresent()).toBe(false);
+    expect(firstElementByTestID('receiver-routing-labels-editor').isPresent()).toBe(false);
+    expect(monitoringView.saveButton.isEnabled()).toBe(false);
+
+    expect(firstElementByTestID('receiver-name').isPresent()).toBe(true);
+    await firstElementByTestID('receiver-name').sendKeys('MyReceiver');
+
+    expect(firstElementByTestID('dropdown-button').isDisplayed()).toBe(true);
+    await firstElementByTestID('dropdown-button').click();
+    await crudView.isLoaded();
+
+    expect(firstElementByTestID('dropdown-menu').isDisplayed()).toBe(true);
+    await firstElementByTestID('dropdown-menu').click();
+    await crudView.isLoaded();
+
+    // these should be shown after receiverType selected
+    expect(firstElementByTestID('pagerduty-receiver-form').isPresent()).toBe(true);
+    expect(firstElementByTestID('receiver-routing-labels-editor').isPresent()).toBe(true);
+
+    expect(firstElementByTestID('pagerduty-key-label').getText()).toEqual('Routing Key');
+    await firstElementByTestID('integration-type-prometheus').click();
+    expect(firstElementByTestID('pagerduty-key-label').getText()).toEqual('Service Key');
+    await firstElementByTestID('integration-key').sendKeys('<integration_key>');
+
+    expect(monitoringView.saveButton.isEnabled()).toBe(true); // should be enabled at this point
+
+    await firstElementByTestID('label-name-0').sendKeys('severity');
+    await firstElementByTestID('label-value-0').sendKeys('warning');
+
+    await monitoringView.saveButton.click();
+
+    await crudView.isLoaded();
+
+    monitoringView.getFirstRowAsText().then((text) => {
+      expect(text).toEqual('MyReceiver pagerduty severity = warning');
+    });
+  });
+
+  it('edits PagerDuty Receiver correctly', async () => {
+    await crudView.isLoaded();
+    expect(crudView.resourceRows.count()).toBe(2);
+    await monitoringView.clickFirstRowKebabAction('Edit Receiver');
+    await browser.wait(until.presenceOf(firstElementByTestID('cancel')));
+    expect(firstElementByTestID('receiver-name').getAttribute('value')).toEqual('MyReceiver');
+    expect(firstElementByTestID('dropdown-button').getText()).toEqual('PagerDuty');
+    expect(firstElementByTestID('pagerduty-key-label').getText()).toEqual('Service Key');
+    expect(firstElementByTestID('integration-key').getAttribute('value')).toEqual(
+      '<integration_key>',
+    );
+    expect(firstElementByTestID('label-name-0').getAttribute('value')).toEqual('severity');
+    expect(firstElementByTestID('label-value-0').getAttribute('value')).toEqual('warning');
+
+    // Edit Values
+
+    await firstElementByTestID('receiver-name').clear();
+    await firstElementByTestID('receiver-name').sendKeys('MyEditedReceiver');
+    await firstElementByTestID('label-name-0')
+      .clear()
+      .then(() => firstElementByTestID('label-name-0').sendKeys('cluster'));
```

Comment:
We shouldn't mix async/await and promises (here and just below).

```ts
await firstElementByTestID('label-name-0').clear();
await firstElementByTestID('label-name-0').sendKeys('cluster');
```

Reply:
I was has having timing issues, clear() is a promise.  Basically sendKeys() was executing before clear(), just in these cases, I know it works elsewhere

Reply:
> I was has having timing issues, clear() is a promise.

`await` waits for the promise to resolve